### DISCUSSION TITLE
Zigbee2mqtt fails to start. Cannot read property 'length' of null at AdapterNvMemory.init

### DISCUSSION THREAD

**Discussion by john-beck** [Koenkk/zigbee2mqtt#9767] [General]
Hello,

I am attempting to run zibee2mqtt (1.22.0-3) on a RPI3b+ running HA (core-2021.11.4 & HAOS 6.6).

The zigbee adapter I am using is the slae.sh CC2652RB.

This RPI and stick were working at one point but I needed to rebuild the system for other reasons and now I am getting errors that I cannot figure out. I have reinstalled zigbee2mqtt several times and I also tried zigbee2mqtt edge. Neither works.

HA does see the CC2652RB as it recognized it a a "new discovered device". 

I have had no luck finding any information related to the errors I see in the zigbee2mqtt logs. BTW - the Mosquitto MQTT broker is running and appears to be working. There are no log errors for that integration.

on the webUI the error is:
502: Bad Gateway

**in the zigbee2mqtt logs the errors are:**
Zigbee2MQTT:error 2021-11-20 11:59:51: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-11-20 11:59:51: Failed to start zigbee
Zigbee2MQTT:error 2021-11-20 11:59:51: Check https://www.zigbee2mqtt.io/information/FAQ.html#help-zigbee2mqtt-fails-to-start for possible solutions
Zigbee2MQTT:error 2021-11-20 11:59:51: Exiting...
Zigbee2MQTT:error 2021-11-20 11:59:51: TypeError: Cannot read property 'length' of null
    at AdapterNvMemory.init (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/adapter-nv-memory.ts:28:42)
    at ZnpAdapterManager.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:60:9)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:123:29)
    at Zigbee.start (/app/lib/zigbee.ts:58:27)
    at Controller.start (/app/lib/controller.ts:100:27)
    at start (/app/index.js:97:5)
[11:59:52] INFO: Handing over control to Zigbee2mqtt Core ...
> zigbee2mqtt@1.22.0 start
> node index.js

**In the HA logs the related errors are:**
21-11-20 11:46:00 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.2:8099 ssl:default [Connect call failed ('172.30.33.2', 8099)]
21-11-20 11:46:33 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.2:8099 ssl:default [Connect call failed ('172.30.33.2', 8099)]
21-11-20 11:46:34 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.2:8099 ssl:default [Connect call failed ('172.30.33.2', 8099)]
21-11-20 11:46:35 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.2:8099 ssl:default [Connect call failed ('172.30.33.2', 8099)]
21-11-20 11:46:41 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.2:8099 ssl:default [Connect call failed ('172.30.33.2', 8099)]
21-11-20 11:46:55 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.2:8099 ssl:default [Connect call failed ('172.30.33.2', 8099)]

**Here is my zigbee2mqtt config:** 
data_path: /config/zigbee2mqtt
external_converters: []
devices: devices.yaml
groups: groups.yaml
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://core-mosquitto:1883
serial:
  port: >-
    /dev/serial/by-id/usb-Silicon_Labs_slae.sh_cc2652rb_stick_-_slaesh_s_iot_stuff_00_12_4B_00_22_98_A8_AC-if00-port0
advanced:
  log_level: warn
  pan_id: 6754
  channel: 11
  network_key:
    - 1
    - 3
    - 5
    - 7
    - 9
    - 11
    - 13
    - 15
    - 0
    - 2
    - 4
    - 6
    - 8
    - 10
    - 12
    - 13
  availability_blocklist: []
  availability_passlist: []
device_options: {}
blocklist: []
passlist: []
queue: {}
frontend:
  port: 8099
experimental: {}
availability: false
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: '-d -d'
  log: false

**Comment by john-beck** [comment]
Just adding, in the zigbee2mqtt error log, the directory path referencing subdirectories and files under /app/node_modules/zigbee-herdsman/**src/** does not exist. In the zigbe2mqtt docker container there is a /app/node_modules/zigbee-herdsman/**dist** path that has the referenced subdirectories and files in it.

I wondered if maybe there was a configuration issue so I wanted to test the theory by creating a symbolic link to dist from src which I can do in the docker container.  But since I can't make permanent changes, the link is wiped out when I restart zigbee2mqtt which restarts the container. I guess I'm stuck at this point.

**Comment by francisp2** [comment]
You will need to re-flash your stick.

**Reply by john-beck** [reply]
I'll give it a try but I'm curious, what happens that causes the need to re-flash the stick?

**Comment by francisp2** [comment]
https://github.com/Koenkk/zigbee2mqtt/issues/8504

**Reply by john-beck** [reply]
Thanks for the link.

**Comment by john-beck** [comment]
Following francisp2's advice, I flashed the slae.sh CC2652RB.

For reference, I flashed the stick on a Windows machine using the TI SmartRF Flash Programmer 2. The software is free to use from TI but you have to create an account to download it.
https://www.ti.com/tool/download/FLASH-PROGRAMMER-2

**>>>** I needed to download and install this driver before the software could see the Zigbee stick:
https://www.silabs.com/developers/usb-to-uart-bridge-vcp-drivers

I used the the driver called: "CP210x Universal Windows Driver." The instructions to install it are in CP210x_Universal_Windows_Driver_ReleaseNotes.txt

For what to do in the software I followed slaesh's instructions at the link below. Just match your settings in the application with his the screen shot.
https://slae.sh/projects/cc2652/#smart-rf-flash-programmer2

I set the buttons as he instructs to active bootloader mode. This instruction was vague: "The bootloader-pin can be released." I assumed he meant bootloader button. 

Anyway, I was able to flash the stick and Zigbee2mqtt now starts and all my Zigbee devices are working.

Thanks for pointing my in the right direction francisp2.

**Comment by jim82522** [comment]
For anyone encountering a similar issue, especially with Sonoff ZBDongle-P (CC2652P) or ZBDongle-E (EFR32MG21) devices, there’s now a much simpler solution available.

🛠️ Sonoff has released a web-based “Quick Flasher” tool, which makes reflashing much easier and doesn’t require you to install complex software—just the [CP210x Universal Windows Driver](https://www.silabs.com/developers/usb-to-uart-bridge-vcp-drivers).

🔗 Try it here: [Sonoff Dongle Quick Flasher](https://dongle.sonoff.tech/sonoff-dongle-quick-flasher)

This is a convenient alternative to using TI’s Flash Programmer and should be especially helpful for Windows users who just want to get their Zigbee stick back up and running without hassle.
